generate-examples/src/certs-to-x5c.js

```diff
@@ -46,31 +52,40 @@ const DERtoJWK = async (key, certs) => {
             // SHA-256 kid auto-generated by key import
         }
     );
-    return {
-        private: JSON.stringify(jwk.toJSON(true)),
-        public:  JSON.stringify(jwk.toJSON(false))
+    return [JSON.stringify(jwk.toJSON(true)), JSON.stringify(jwk.toJSON(false))]
+}
+
+const main = async (options) => {
+    if (!options.key || !options.cert) {
+        console.log("Missing --key or --cert argument");
+        program.help();
+    }
+    try {
+        // open or create public/private key stores
+        const getStore = async path => fs.existsSync(path)  ? await jose.JWK.asKeyStore(JSON.parse(fs.readFileSync(path)))  : jose.JWK.createKeyStore();
+        const store = [await getStore(options.private), await getStore(options.private)];
+
+        // read and convert input key and certs to DER format
+        const keyFile = PEMtoDER(fs.readFileSync(options.key,'UTF-8'));
+        const certFiles = options.cert.map(certPath => PEMtoDER(fs.readFileSync(certPath,'UTF-8')));
+
+        // convert DER key to JWK, adding a x5c value with cert chain
+        keys = await DERtoJWK(keyFile, certFiles);
+        
+        // output public/private key stores with new key
+        keys.map( async (k,i) => {

```

Comment:
It's not clear to me that this work will always finish before the program exits. I think you may need to `await Primise.all()` on the array of promises from this map, or something equivalent. (Or if this is happening implicitly, I want to make sure I understand what is enforcing this.)

Reply:
Pushed update